MetaMask/metamask-extension
# feat: Adding "cookie id" to metrics event

## **Description**
The PR sets up a stream communication with the label 'metamask-cookie-handler'. The data sent from the page will be intercepted at the content-script and forwarded to the extension. The 'ga_client_id' sent from the page is saved as `marketingCampaignCookieId` in the metaMetricsController state and fetched when building the context for each event.



[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/26697?quickstart=1)

## **Related issues**

Part of https://github.com/MetaMask/MetaMask-planning/issues/2916

## **Manual testing steps**

To test the stream, Open https://metamask.io/ , and write a request in dev tools console with the following `window.postMessage`  would do. This should update the metametricsController state with `marketingCampaignCookieId`. Logging the payload in `_track()` in MetaMetricsController will show that the `marketingCampaignCookieId` is added to the context.

```
window.postMessage(
      {
        target: 'metamask-contentscript', 
        data: {
          name: 'metamask-cookie-handler', 
          data: {
            jsonrpc: '2.0',
            method: 'getCookieFromMarketingPage',
            params: [{ ga_client_id: 12345, origin: 'https://metamask.io' }],
            id: '1234',
            origin: 'https://metamask.io',
          },
        },
      },
      'https://metamask.io',
    );
```
Test the track event sending after this has this id in its context as `marketingCampaignCookieId`.
## **Screenshots/Recordings**



### **After**
Loading a website in the from the white list:

https://github.com/user-attachments/assets/b36b7786-b946-4da1-8424-c0661e6f3d49

Initially loading a website not from the whitelist and then loading a website from whitelist to compare the context data:


https://github.com/user-attachments/assets/a49fd520-7938-4114-88ac-27bbef4c217f


## **Pre-merge author checklist**






## **Pre-merge reviewer checklist**
